common/src/View/RepeatStack.h

```diff
@@ -0,0 +1,75 @@
+/*
+ Copyright (C) 2020 Kristian Duske
+
+ This file is part of TrenchBroom.
+
+ TrenchBroom is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ TrenchBroom is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with TrenchBroom. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TrenchBroom_RepeatStack
+#define TrenchBroom_RepeatStack
+
+#include <functional>
+#include <vector>
+
+namespace TrenchBroom {
+    namespace View {
+        class RepeatStack {
+        public:
+            using RepeatableAction = std::function<void()>;
+        private:
+            std::vector<RepeatableAction> m_stack;
+            bool m_clearOnNextPush;
+            mutable bool m_repeating;
+        public:
+            /**
+             * Creates a new instance.
+             */
+            RepeatStack();
+
+            /**
+             * Returns the number of repeatable actions on this repeat stack.
+             */
+            size_t size() const;
+
+            /**
+             * Adds the given repeatable action to this repeat stack.
```

Comment:
If we're going to say anything here, it needs to break down all 3 behaviours:

1. Does nothing if currently repeating.
2. If the m_clearOnNextPush flag was set, clears the repeat stack and clears the flag
3. Pushes the given action

Reply:
ok

Reply:
https://github.com/TrenchBroom/TrenchBroom/pull/3501/commits/cc3876b94e385d3b80e3621ece753cf90db52ada